provider/runbook_resource.go

```diff
@@ -244,11 +258,17 @@ func updateResourceFireHydrantRunbook(ctx context.Context, d *schema.ResourceDat
 	for _, currentStep := range steps {
 		step := currentStep.(map[string]interface{})
 
+		if step["repeats"].(bool) == true && step["repeats_duration"].(string) == "" {
+			return diag.Errorf("Error updating runbook, step repeats requires repeat_duration to be set")
+		}
```

Comment:
i think it would be good to check for the inverse here as well, that repeats_duration should not be set if repeats is set to false. can also simplify the error message here to focus on just the argument error
```suggestion
		if step["repeats"].(bool) == true && step["repeats_duration"].(string) == "" {
			return diag.Errorf("step repeats requires step repeats_duration to be set")
		}
		if step["repeats"].(bool) == false && step["repeats_duration"].(string) != "" {
			return diag.Errorf("step repeats_duration required step repeat to be set to true")
		}
```